### PR TITLE
feat: CSV File Upload form updates

### PR DESCRIPTION
**Pull request by Antonio-RiveroMartnez**
### SUMMARY
This is going to be the first of a set of PRs to improve the experience for our users in the CSV file upload form. Here we are reordering some fields of the form and changing their Name and descriptions.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:
https://user-images.githubusercontent.com/38889534/198689750-2996d449-af67-43f6-b75f-94c79c913960.mov

After:
https://user-images.githubusercontent.com/38889534/198690113-03a711d0-f903-484e-a844-cae9fda8a8eb.mov

### TESTING INSTRUCTIONS
1. Open the CSV file upload form
2. Check the form matches the proposed changes

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [X] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/views/database/forms.py [hunk 1]
```diff
@@ -133,129 +124,136 @@ class CsvToDatabaseForm(UploadToDatabaseForm):
...
     )
-    index = BooleanField(
-        _("Dataframe Index"), description=_("Write dataframe index as a column.")
+    null_values = JsonListField(
+        _("Null Values"),
+        default=config["CSV_DEFAULT_NA_NAMES"],
+        description=_(
+            "Json list of the values that should be treated as null. "
+            'Examples: [""], ["None", "N/A"], ["nan", "null"]. '
+            "Warning: Hive database supports only single value"
```
Comment (eschutho) [inline superset/views/database/forms.py:198]: nit typo in existing text.. should this be "only a single value"?

Comment (Antonio-RiveroMartnez) [reply superset/views/database/forms.py]: Oh Thanks! Fixed

**Review by eschutho — APPROVED** [review]
Code looks great. Will defer to @jinghua-qa or @yousoph for visual review.

**Comment by andrey-zayats** [comment]
LGTM!

**Pull request merged**